Products/Zing/tests/test_datamaps.py

```diff
@@ -383,6 +380,17 @@ def getDeviceClassName(self):
     def getProductionStateString(self):
         return 'test_production_state_string'
 
+    def getPrimaryParent(self):
```

Comment:
this doesn't allow testing of nested components, or testing of handling relations on the 'os' and 'hw' attributes of devices.